### PR TITLE
Repurpose Get/SetConfig as import/export support

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Repurpose Get/SetConfig as import/export support

## Motivation and Context
To be able to export/import entire server configs

## How to test this PR?
https://github.com/minio/mc/pull/2974 new `mc` PR to test this

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#509d6eb6-f48c-4e47-9fee-20b5f4f14809) |
| mint-large-bucket.sh | :x: [more...](#ab08ab1e-36cd-4f4c-8f77-0a26693403ae) |
<a name="509d6eb6-f48c-4e47-9fee-20b5f4f14809"></a>
### 8578-81e2ea0/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30163
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2a0ad9638763:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-09s5waf3cj0vcb",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364689,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15DAB21D9693527F, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="ab08ab1e-36cd-4f4c-8f77-0a26693403ae"></a>
### 8578-81e2ea0/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:32212
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 03da78ea47c4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 47 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 2 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 11 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 8 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 36 seconds
{
  "args": {},
  "duration": 89467,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8578-81e2ea0

**Review by nitisht — CHANGES_REQUESTED** [review]
~While testing, `mc` returns empty output~

Nevermind, it happens when config is empty

**Review by nitisht — APPROVED** [review]

File: cmd/admin-handlers-config-kv.go [hunk 1]
```diff
@@ -422,17 +424,28 @@ func (a adminAPIHandlers) SetConfigHandler(w http.ResponseWriter, r *http.Reques
...
 
-	var cfg config.Config
-	if err = json.Unmarshal(configBytes, &cfg); err != nil {
-		logger.LogIf(ctx, err)
-		writeCustomErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrAdminConfigBadJSON), err.Error(), r.URL)
+	cfg := newServerConfig()
+	scanner := bufio.NewScanner(bytes.NewReader(kvBytes))
+	for scanner.Scan() {
+		// Skip any empty lines, or comment like characters
+		if scanner.Text() == "" || strings.HasPrefix(scanner.Text(), config.KvComment) {
```
Comment (Praveenrajmani) [inline cmd/admin-handlers-config-kv.go:438]: Shouldn't we skip `\n` instead of ""?

Comment (harshavardhana) [reply cmd/admin-handlers-config-kv.go:438]: Don't you know the purpose of bufio.Scan?  @Praveenrajmani

Comment (Praveenrajmani) [reply cmd/admin-handlers-config-kv.go:438]: Sorry.. I used `fmt.Println()` in the test code which looked like double lines.. Verified with `fmt.Print`

**Review by Praveenrajmani — APPROVED** [review]
Tested and LGTM

**Pull request merged**